targets/TARGET_NUVOTON/TARGET_M480/crypto/crypto-misc.cpp

```diff
@@ -19,26 +19,46 @@
 #include "mbed_assert.h"
 #include "mbed_critical.h"
 #include "mbed_error.h"
+#include "cmsis_os2.h"
+#include "mbed_rtos_storage.h"
+#include <string.h>
 #include <limits.h>
 #include "nu_modutil.h"
 #include "nu_bitutil.h"
 #include "crypto-misc.h"
+#include "SingletonPtr.h"
+#include "Mutex.h"
+
+/* Consideration for choosing proper synchronization mechanism
+ *
+ * 1. Except for SHA AC which doesn't support context save & restore, we lock all crypto AC
+ *    for just their real operation rather than the whole lifetime of their crypto context.
+ *    For SHA AC, we provide SHA S/W fallback when SHA AC is not available. This policy can
+ *    avoid deadlock. We choose mutex to synchronize access to crypto non-SHA AC. Known
+ *    drawback of this choice is priority inversion.
+ * 2. SHA context can be init'ed in one thread and free'ed in another thread. We cannot
+ *    choose mutex for locking SHA AC for the whole lifetime of SHA context because mutex
+ *    requires lock/unlock in the same thread. We choose atomic flag to synchronize access
+ *    to crypto SHA AC. By just try-acquire SHA AC and fallback to S/W on failure, we avoid
+ *    busy-wait loop which would bite CPU.
+ */
+
+/* Mutex for crypto AES AC management */
+SingletonPtr<rtos::Mutex> crypto_aes_mutex;
```

Comment:
These should still be `static`.

Reply:
@kjbracey-arm Fixed in https://github.com/ARMmbed/mbed-os/pull/9081/commits/2cc76eba9861b631fddc250739da5a5d2f353ff7.